### PR TITLE
synth_xilinx: error out if tristate without '-iopad'

### PR DESCRIPTION
**Pull request by eddiehung**

### CODE REVIEW

File: techlibs/xilinx/synth_xilinx.cc [hunk 1]
```diff
@@ -306,6 +306,10 @@ struct SynthXilinxPass : public ScriptPass
 			run("proc");
 			if (flatten || help_mode)
 				run("flatten", "(with '-flatten')");
+			active_design->scratchpad_unset("tribuf.added_something");
+			run("tribuf -logic");
+			if (!do_iopad && active_design->scratchpad_get_bool("tribuf.added_something"))
+				log_error("Tristate buffers are unsupported without the '-iopad' option.\n");
 			run("tribuf -logic");
```
Comment (mwkmwkmwk) [inline techlibs/xilinx/synth_xilinx.cc]: We already run `tribuf -logic` here...

Comment (eddiehung) [reply techlibs/xilinx/synth_xilinx.cc]: Oooh thanks for spotting my sloppy mistakes!

File: techlibs/xilinx/synth_xilinx.cc [hunk 2]
```diff
@@ -306,6 +306,10 @@ struct SynthXilinxPass : public ScriptPass
 			run("proc");
 			if (flatten || help_mode)
 				run("flatten", "(with '-flatten')");
+			active_design->scratchpad_unset("tribuf.added_something");
+			run("tribuf -logic");
+			if (!do_iopad && active_design->scratchpad_get_bool("tribuf.added_something"))
```
Comment (Xiretza) [inline techlibs/xilinx/synth_xilinx.cc]: This should probably be `if (no_iopad && ...`? `do_iopad` doesn't seem to be defined anywhere.

Comment (eddiehung) [reply techlibs/xilinx/synth_xilinx.cc]: Resolved in 245b8c4a. Thanks!

**Pull request merged**